zk_shell/shell.py

```diff
@@ -1699,25 +1699,14 @@ def fetch(endpoint, states, znodes, ephemerals, datasize, sessions, zxids, idx):
             datasize[idx] = int(dsize)
             sessions[idx] = int(session_count)
 
-            def fetch_zxid(endpoint):
-                zxid = -1
-                try:
-                    stat = self._zk.cmd(hosts_to_endpoints(endpoint), "stat")
-                    for line in stat.split("\n"):
-                        if "Zxid:" in line:
-                            zxid = int(line.split(None)[1], 0)
-                except:
-                    pass
-                return zxid
-
-            # the stat cmd is a bit flaky, so try a few times
-            zxid = -1
-            for i in range(0, stat_retries):
-                zxid = fetch_zxid(endpoint)
-                if zxid != -1:
-                    break
-
-            zxids[idx]= zxid
+            try:
+                srvr = self._zk.cmd(hosts_to_endpoints(endpoint), "srvr")
+                for line in srvr.split("\n"):
+                    if "Zxid:" in line:
+                        zxids[idx] = int(line.split(None)[1], 0)
+                        break
+            except:
+                zxids[idx] = -1
```

Comment:
I think we need `zxids[idx] = -1` before the try/except block; if the server is not serving (e.g.: quorum hasn't been formed yet) `srvr` will reply with something like `Server is not serving requests`.. which means the zxid entry for `idx` would be missing.

Reply:
oh so the except block is not hit and we also don't hit the if block. I can see that. yeah maybe define above the try block to ensure it has a value. Good catch.

Reply:
I pushed a change - but I'm not a python programmer so let me know.